src/creeps/Zombie.cpp

```diff
@@ -1,61 +1,450 @@
+             }
+
+            setPath(path);
+            return path;
+        }
+
+        // traverse neighbors
+        for (i = 0; i < DIR_CAP;i++) {
+            // neighbor coordinates
+            xdx = x + mx[i];
+            ydy = y + my[i];
+
+            // not evaluated & not outside (bound checking)
+            if (!(xdx < 0 || xdx > col -1 || ydy < 0 || ydy > row - 1
+                || map[xdx][ydy] == 1 || closedNodes[xdx][ydy] == 1)) {
+
+                // generate a child node
+                childNode = Node(xdx, ydy, curNode.getLevel(), curNode.getPriority());
+                childNode.nextLevel(i);
+                childNode.updatePriority(xNodeDest, yNodeDest);
+
+                // if it is not in the open list then add into that
+                if (openNodes[xdx][ydy] == 0) {
+                    openNodes[xdx][ydy] = childNode.getPriority();
+                    pq[index].push(childNode);
+                    // update the parent direction info
+                    dirMap[xdx][ydy] = (i + DIR_CAP/2)%DIR_CAP;
+                } else if (openNodes[xdx][ydy] > childNode.getPriority()) {
+                    // update the priority info
+                    openNodes[xdx][ydy]= childNode.getPriority();
+                    // update the parent direction info
+                    dirMap[xdx][ydy] = (i + DIR_CAP/2)%DIR_CAP;
+
+                    // use a queue and a backup queue to put the best node (with highest priority)
+                    // on the top of the queue, which can be chosen later on to build the path.
+                    while (!(pq[index].top().getXPos() == xdx &&
+                           pq[index].top().getYPos() == ydy)) {
+                        pq[1-index].push(pq[index].top());
+                        pq[index].pop();
+                    }
+
+                    pq[index].pop();
+
+                    // switch to pq with smaller size
+                    if (pq[index].size() > pq[1-index].size()) {
+                        index = 1 - index;
+                    }
+
+                    while (!pq[index].empty()) {
+                        pq[1-index].push(pq[index].top());
+                        pq[index].pop();
+                    }
+                    index = 1 - index;
+                    pq[index].push(childNode);
+                }
+            }
+        }
+    }
+
+    return ""; // no route found
 }
 
-void Zombie::generateRandomMove(){
-    double cosVal = cos(getAngle()*PI/180.0);
-    double sinVal = sin(getAngle()*PI/180.0);
+/**
+ * Check to see if the zombie is still within the screen bounds before moving
+ * Fred Yang
+ * Feb 14
+ */
+bool Zombie::checkBounds(const float x, const float y) const {
+    return (!(x < 0 || x > MAP_WIDTH || y < 0 || y > MAP_HEIGHT));
+}
 
-    const int x = getVelocity();
-    const int y = getVelocity();
+/**
+ * Check to see if two squares overlapped
+ */
+bool Zombie::overlapped(const float x1, const float y1,
```

Comment:
all collisions without exception should use the collision handlers as they use quadtrees